Tests/DatadogTests/Datadog/Core/Upload/DataUploadWorkerTests.swift

```diff
@@ -22,68 +39,187 @@ class DataUploadWorkerTests: XCTestCase {
     }
 
     func testItUploadsAllData() throws {
-        let dateProvider = RelativeDateProvider(advancingBySeconds: 1)
-        let orchestrator = FilesOrchestrator(
-            directory: temporaryDirectory,
-            performance: StoragePerformanceMock.writeEachObjectToNewFileAndReadAllFiles,
-            dateProvider: dateProvider
-        )
-        let writer = FileWriter(
-            dataFormat: .mockWith(prefix: "[", suffix: "]"),
-            orchestrator: orchestrator,
-            queue: fileReadWriteQueue
-        )
-        let reader = FileReader(
-            dataFormat: .mockWith(prefix: "[", suffix: "]"),
-            orchestrator: orchestrator,
-            queue: fileReadWriteQueue
-        )
         let server = ServerMock(delivery: .success(response: .mockResponseWith(statusCode: 200)))
         let dataUploader = DataUploader(
             urlProvider: .mockAny(),
             httpClient: HTTPClient(session: .serverMockURLSession),
             httpHeaders: .mockAny()
         )
-
         // Write 3 files
         writer.write(value: ["k1": "v1"])
         writer.write(value: ["k2": "v2"])
         writer.write(value: ["k3": "v3"])
+        // Start logs uploader
+        try withExtendedLifetime(
+            DataUploadWorker(
+                queue: uploaderQueue,
+                fileReader: reader,
+                dataUploader: dataUploader,
+                uploadConditions: DataUploadConditions.alwaysUpload(),
+                delay: DataUploadDelay(performance: UploadPerformanceMock.veryQuick),
+                featureName: .mockAny()
+            )
+        ) {
+            let recordedRequests = server.waitAndReturnRequests(count: 3)
+            XCTAssertTrue(recordedRequests.contains { $0.httpBody == #"[{"k1":"v1"}]"#.utf8Data })
+            XCTAssertTrue(recordedRequests.contains { $0.httpBody == #"[{"k2":"v2"}]"#.utf8Data })
+            XCTAssertTrue(recordedRequests.contains { $0.httpBody == #"[{"k3":"v3"}]"#.utf8Data })
 
+            uploaderQueue.sync {} // wait until last "process upload" operation completes (to make sure "delete file" was requested)
+            fileReadWriteQueue.sync {} // wait until last scheduled "delete file" operation completed
+
+            XCTAssertEqual(try temporaryDirectory.files().count, 0)
+        }
+    }
+
+    // swiftlint:disable multiline_arguments_brackets
+    func test_whenThereIsNoBatch_thenIntervalIncreases() throws {
```

Comment:
Convention: we commonly use `testWhen`, not `test_when`

<img width="273" alt="Screenshot 2020-09-11 at 17 33 40" src="https://user-images.githubusercontent.com/2358722/92944996-fa2e7400-f454-11ea-8741-9d611a356ea2.png">